pytorch_lightning/plugins/training_type/tpu_spawn.py

```diff
@@ -312,7 +312,17 @@ def save_checkpoint(self, checkpoint: Dict[str, Any], filepath: _PATH) -> None:
             checkpoint: dict containing model and trainer state
             filepath: write-target file's path
         """
-        return self.checkpoint_io.save_checkpoint(checkpoint, filepath)
+        # xm.save needs to be called on all ranks. It internally checks if the local rank is 0
+        self.checkpoint_io.save_checkpoint(checkpoint, filepath)
+
+    def remove_checkpoint(self, filepath: _PATH) -> None:
```

Comment:
RFC: should we have checkpoint_io for tpu and deepspeed instead of override functions in strategies directly?

Reply:
TPU: override is just rank logic which should be controlled by strategies right

deepspeed: my thoughts are it would be hard/inadvisable to fit inside `IOPlugin`, since it looks like it involves things like `broadcast` call, checking trainer state, which we ideally shouldn't do in `IOPlugin`. I think we'd need to pass `deepspeed_engine` too
https://github.com/PyTorchLightning/pytorch-lightning/blob/3c4d06bd42ff3282a017cd9fa879bbd68e8568df/pytorch_lightning/plugins/training_type/deepspeed.py#L755-L768